### PR TITLE
DOCSP-43209 - v3.1

### PR DESCRIPTION
**Pull request by mongoKart**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-rust/blob/master/REVIEWING.md)

JIRA - https://jira.mongodb.org/browse/DOCSP-43209
Staging - https://preview-mongodbmongokart.gatsbyjs.io/rust/docsp-43209-v3.1/whats-new/

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

File: source/whats-new.txt [hunk 1]
```diff
@@ -20,12 +20,38 @@ What's New
...
+What's New in 3.1
+-----------------
+
+The {+driver-short+} v3.1 release includes the following features,
+improvements, and fixes:
+
+- Adds support for more than one KMS (Key Management Service) provider of the same type.
+  To learn more about KMS providers, see
+  :manual:`Queryable Encryption </core/queryable-encryption/>` in the {+server+}
+  manual.
```
Comment (rustagir) [inline source/whats-new.txt:41]: S; reverse order of phrase and abbreviation

File: source/whats-new.txt [hunk 2]
```diff
@@ -20,12 +20,38 @@ What's New
...
+- Adds support for v2 of the Queryable Encryption range protocol. Range indexes for
+  Queryable Encryption are now GA.
+- Adds support for the ``type`` field when creating indexes. To learn more about
+  creating indexes, see :manual:`Indexes </indexes/>` in the {+server+} manual.
+- Implements the Happy Eyeballs algorithm for managing
+  TCP connections. To learn more about this algorithm, see the Rust documentation
+  for the `happy_eyeballs <https://docs.rs/happy_eyeballs/0.2.0/happy_eyeballs/>`__ crate.
+- The driver now publishes a ``TopologyDescriptionChangedEvent`` event when a topology
+  closes.
```
Comment (rustagir) [inline source/whats-new.txt:50]:
```suggestion
- The driver publishes a ``TopologyDescriptionChangedEvent`` event when a topology
  closes.
```

**Review by rustagir — APPROVED** [review]
lgtm with small suggestions

File: source/whats-new.txt (hunk 1 above)
Comment (mongoKart) [reply source/whats-new.txt:41]: good call

**Review by abr-egn — APPROVED** [review]

**Pull request merged**